### PR TITLE
Add Solana claim flow

### LINKED ISSUE #28030: mobile: sol staking – claiming
## Objective

Implement the claim flow for mobile Trezor Suite, allowing users to claim their assets after the deactivation period (epoch) has completed. The flow covers the entry from the staking dashboard's "Claim ready" banner, a claim form showing the amount and fee, Trezor device signing, and the post-transaction result.

## Acceptance criteria

- Entry Point
  - User accesses the claim flow from the staking management dashboard.
  - The dashboard shows a "Pending actions" section with a claim-ready banner:
    - Check icon + text: indicating unstaked amount ready to be claimed, green "Claim" CTA button
- Claim Form
  - Headline describing the flow (e. g. "Claim SOL") with an X close button.
  - Amount display: Shows the claimable amount (e.g., "1.24580213 SOL / $211.79"). This is not editable — the user claims the full available amount.
  - Fee display: Shows the estimated network fee (e.g., "0.000005 SOL / ~$0.001").
    - Informational note: "You'll claim the SOL instantly" with a checkmark, confirming that once the claim transaction is confirmed, the SOL will be available immediately.
  - "Review & Sign" green CTA button at the bottom.
  - Existing copy from the firmware/desktop Suite for this screen is to be reused. 
- Signing — "Continue on your Trezor"
  - "Continue on your Trezor" screen with green progress stepper.
  - Action: "Claim"
    -  Description: "Claim SOL from account?" (Note: the selected network should be referred here), Total including fee.
    - Amount and maximum fee breakdown.
  - Final breakdown summary with "Claim now" CTA
- Transaction Result — "Claim transaction"
  - "Claim transaction" detail screen showing: Upward arrow icon with "Pending" badge, Claimed amount (e.g., "+ 0.85234712 SOL") with fiat equivalent (e.g., "≈$144.90"), Fee, date, from/to addresses.
  - closing by X button takes the user back to the staking dashboard

### PR DESCRIPTION
**Pull request by myasiura-stack**
## Description

Adds the Solana staking claim flow to the native (mobile) app, reusing the existing network-agnostic claim screens with Solana-specific transaction composition, signing, and amount handling. For now, the two Solana claim entry points — the "Your stakes" sheet on the Earn dashboard and the staking dashboard's claimable card — open the "Manage staking in Trezor Suite for desktop" info modal instead of starting the flow, so users are directed to desktop until claiming is enabled on mobile. Ethereum claiming is unaffected.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/28030

## Screenshots:
Simulator Screenshot - iPhone 16 Plus - 2026-06-05 at 11 22 34

### CODE REVIEW

File: suite-native/module-earn/src/components/EarnAccountCard.tsx [hunk 1]
```diff
@@ -64,7 +67,10 @@ export const EarnAccountCard = ({ item, onPress, onClaimPress }: EarnAccountCard
     const { applyStyle } = useNativeStyles();
     const isStakingItem = item.type === 'staking';
     const isStablecoinYieldItem = item.type === 'stablecoin-yield';
-    const isSupportedStaking = isStakingItem && isSupportedEthStakingNetworkSymbol(item.symbol);
+    const isSupportedStaking =
+        isStakingItem &&
+        (isSupportedEthStakingNetworkSymbol(item.symbol) ||
+            isSupportedSolStakingNetworkSymbol(item.symbol));
```
Comment (53gur0) [inline suite-native/module-earn/src/components/EarnAccountCard.tsx]: can we just call it isSupportedStaking and then figure out all the details inside the function?

**Comment by BrantalikP** [comment]
@coderabbitai review

File: suite-native/module-earn/src/components/StakeClaimableCard.tsx [hunk 2]
```diff
@@ -52,53 +64,66 @@ export const StakeClaimableCard = ({ accountKey }: StakeClaimableCardProps) => {
...
+
         navigation.navigate(RootStackRoutes.ClaimReview, { accountKey, symbol });
-    }, [accountKey, navigation, symbol, isClaimingDisabled]);
+    }, [accountKey, navigation, symbol, isClaimingDisabled, openInfoModal]);
 
     if (
         !symbol ||
         !isPositiveBalance(claimableAmount) ||
-        !isSupportedEthStakingNetworkSymbol(symbol)
+        !(isSupportedEthStakingNetworkSymbol(symbol) || isSupportedSolStakingNetworkSymbol(symbol))
```
Comment (53gur0) [inline suite-native/module-earn/src/components/StakeClaimableCard.tsx]: same here if we'd be changing stacking support to a higher level function

File: suite-native/staking/src/__tests__/stakeFormSolanaNativeThunks.test.ts [hunk 3]
```diff
@@ -312,4 +323,58 @@ describe('signSolanaStakingTransactionNativeThunk', () => {
...
+        expect(prepareClaimSolTxMock).toHaveBeenCalledTimes(1);
+        expect(prepareStakeSolTxMock).not.toHaveBeenCalled();
+        expect(solanaSignTransactionMock).toHaveBeenCalledTimes(1);
+        expect(solanaTxShim.addSignature).toHaveBeenCalledWith(
+            solAccount.descriptor,
+            'ed25519signature',
+        );
+    });
+
+    it('claims even when the precomposed amount is zero, since claim ignores the amount', async () => {
```
Comment (53gur0) [inline suite-native/staking/src/__tests__/stakeFormSolanaNativeThunks.test.ts]: What would be a point of that to the user? Sounds like a negative invariant for me rather than passing test

File: suite-native/module-earn/src/screens/ClaimReviewScreen.tsx [hunk 4]
```diff
@@ -65,15 +63,18 @@ export const ClaimReviewScreen = () => {
...
+
+        const calldata = account.networkType === 'ethereum' ? buildClaimWithdrawRequestData() : '';
+        const amount = account.networkType === 'solana' ? (claimableAmount ?? '0') : '0';
+
+        return buildEarnComposeFormState(
+            getStakingContractAddress(account, 'claim'),
+            amount,
+            calldata,
+        );
+    }, [account, claimableAmount]);
```
Comment (53gur0) [inline suite-native/module-earn/src/screens/ClaimReviewScreen.tsx]: nit: two distinct values checks could be more readable here to not confuse reader why solana has calldata and why it's empty

**Review by BrantalikP — APPROVED** [review]

File: suite-native/staking/src/__tests__/stakeFormSolanaNativeThunks.test.ts (hunk 3 above)
Comment (myasiura-stack) [reply suite-native/staking/src/__tests__/stakeFormSolanaNativeThunks.test.ts]: Good catch! The claim is intentional: it claims all rewards on-chain and never uses a composed amount, unlike stake/unstake, which require a non-zero amount. I renamed the test and added an assertion so that the contract is explicit, not read as an edge case.
